### PR TITLE
chore(nodebuilder): timeouts, errors and logs

### PR DESCRIPTION
**Pull request by Wondertan**
Was originally for debugging purposes but actually solved the issue. The bootstrapper with a corrupted DAGstore state was not able to start quickly enough. 

Ideally, we should make this value configurable(https://github.com/celestiaorg/celestia-node/issues/2126)

### CODE REVIEW

**Comment by Wondertan** [comment]
So this actually fixed bootstrapper, as it didn't have enough time to Start.

**Review by renaynay — DISMISSED** [review]

File: nodebuilder/node.go [hunk 1]
```diff
@@ -149,7 +161,11 @@ func (n *Node) Stop(ctx context.Context) error {
 func newNode(opts ...fx.Option) (*Node, error) {
 	node := new(Node)
 	app := fx.New(
-		fx.NopLogger,
+		fx.WithLogger(func() fxevent.Logger {
+			zl := &fxevent.ZapLogger{Logger: fxLog.Desugar()}
+			zl.UseLogLevel(zapcore.DebugLevel)
+			return zl
+		}),
```
Comment (walldiss) [inline nodebuilder/node.go:168]: It is unclear from the PR, what component would take  `fxevent.Logger` as dependancy?  Also would you consider fx log level configurable?

File: nodebuilder/node.go [hunk 2]
```diff
@@ -134,11 +143,14 @@ func (n *Node) Stop(ctx context.Context) error {
 
 	err := n.stop(ctx)
 	if err != nil {
-		log.Errorf("Stopping %s Node: %s", n.Type, err)
-		return err
+		log.Debugf("error stopping %s Node: %s", n.Type, err)
+		if errors.Is(err, context.DeadlineExceeded) {
+			return fmt.Errorf("node: failed to stop within timeout(%s): %w", Timeout, errors.Unwrap(err))
+		}
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply nodebuilder/node.go:149]: Sure, but I don't see any issues with accessing globals in this case. We anyway do that in this method already

File: nodebuilder/node.go (hunk 1 above)
Comment (Wondertan) [reply nodebuilder/node.go:168]: This is a special WithLogger option and it's not a Provide. The given logger is only used by fx itself and not by us. 

We can now control the log verbosity of fx with this change by `fx`.

File: nodebuilder/node.go [hunk 3]
```diff
@@ -27,9 +30,12 @@ import (
 	"github.com/celestiaorg/celestia-node/nodebuilder/state"
 )
 
-const Timeout = time.Second * 15
+var Timeout = time.Second * 30
```
Comment (distractedm1nd) [inline nodebuilder/node.go:33]: The issue is that this number will actually grow with the number of blocks in the DAGstore. It takes quite a while to start already. Maybe we can parallelize this process upstream, or find some other way to let the dagstore initialize before its used by the rest of node

File: nodebuilder/node.go (hunk 1 above)
Comment (distractedm1nd) [reply nodebuilder/node.go:168]: Does this change mean that fx logs will now show up when DEBUG logs are turned on?

Comment (Wondertan) [reply nodebuilder/node.go:168]: Yes

File: nodebuilder/node.go (hunk 3 above)
Comment (Wondertan) [reply nodebuilder/node.go:33]: Thus the issue to make it configurable. Also, keep in mind that it took long on the the node that got kernel panic and had a corrupted DAGStore state

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**